### DISCUSSION TITLE
Does SquareLoss refer to Mean Square Error (MSE) or Sum of Square error over all the training examples?

### DISCUSSION THREAD

**Discussion by yamilbknsu** [fairlearn/fairlearn#1249] [Q&A]
Hello, I'm using the fairlearn package to constrain a regression model for fairness using BoundedLoss constraints. Since I would like all classes to have similar regression error I'm passing a SquareLoss object which takes a min and max argument. My question is related to the values I need to pass to the SquareLoss constructor, I've been assuming that SquareLoss refers to the average square loss across samples but if this is not the case, would it be advisable to input min and max error values as if they were the target min and max mse multiplied by the number of training examples?

Thanks.

**Comment by romanlutz** [comment]
Thanks for reaching out!

Min/max are just for clipping, see https://fairlearn.org/main/user_guide/mitigation/reductions.html#constraints-regression

Does that make sense to you? @yamilbknsu

**Reply by yamilbknsu** [reply]
Thank you for your response.
Yes it does make sense, so during optimization (using `GridSearch` or `ExponentiatedGradient` for example) the error for each individual training example is clipped using the values from `SquareLoss` instead of the objective function for the entire dataset (MSE)?

**Reply by romanlutz** [reply]
I'm not sure I understand the question exactly, but this is the implementation:
https://github.com/fairlearn/fairlearn/blob/476db8c7ca648a4f9f078e91bc3b4058c3a5558a/fairlearn/reductions/_moments/bounded_group_loss.py#L142C9-L145C15

In other words, we clip values to be between the pre-defined min and max from the constructor, and then calculate the square loss of the clipped values. We're not clipping the errors directly, but rather `y_true` and `y_pred`. We're also clipping the aggregate error (MSE or other).
